### DISCUSSION TITLE
Vertical Android Auto support?

### DISCUSSION THREAD

**Discussion by Jirradia** [opencardev/crankshaft#472] [Q&A]
My setup is working flawlessly, yet my issue is the interior in my car puts the display out of reach. it is not so comfortable to use while driving. However, there is a space in the middle for a narrow display. So is there any chance for vertical android auto support in the near future?

**Comment by Jirradia** [comment]
Btw, I've tried
`lcd_rotate=1`
In config.txt but startup terminal keep detecting the rotation and removes it.

**Reply by Royceirving** [reply]
A [fix/thread](https://www.raspberrypi.org/forums/viewtopic.php?t=180407#post_content1147364) that I found was that `lcd_rotate=#` was being ignored by the display driver that I was using. In `/boot/config.txt` , try commenting out the two lines that have `dtoverlay=vc4-kms-v3d`.

**Reply by Jirradia** [reply]
I found this thread before, I tried that command line in multiple ways. It still ignore that screen rotation or intervene to omit the line entirely.

Anyway I found a different single board computer that can run android x86 and from there was able to use android auto. Still work in progress. Thank you for the help, really appreciate it.

**Reply by Royceirving** [reply]
Commenting this anyway because I forgot in my previous reply and feel others might be looking at this. `lcd_rotate` may not do anything either depending on interface you are connecting with. With my setup, I had to use `hdmi_rotate=#` instead.

**Comment by parkerlreed** [comment]
Android Auto itself doesn't support any portrait resolutions. Crankshaft could shove 16:9 into 9:16 but then you would have a tiny area to do anything in.

**Reply by Jirradia** [reply]
Sorry it took me a while to respond. Android support vertical to square-ish display you could look it up. I had two options either to use android auto by cable or android auto for phones app. Anyway, I found a different way to make it vertical. Thank you for the response.

**Reply by boehm-e** [reply]
@Jirradia  how did you do?

**Reply by Jirradia** [reply]
@boehm-e I had to skip crankshaft, I still have my raspberry as a temporary solution.

Just in case you skipped my reply above, I'm in the process of narrowing down my options for other SBC. I have an old Dell laptop at the moment to prototype.
I installed Android x86 (6.0 marshmallow) on my laptop since it's the last version of Android x86 that "support" vertical display. Download a great app called [Head unit reloaded trial](https://play.google.com/store/apps/details?id=gb.xxy.trial&hl=en_US&gl=US). This is the trial version. I'd suggest to buy the [full version](https://play.google.com/store/apps/details?id=gb.xxy.hr&hl=en&gl=US)  once you're happy with whatever you want to accomplish, and believe me it is worth it. The app developer has an amazing support, just read the comments.
You will see a stretched android auto depending on your screen aspect ratio. Here you just have to tweak the settings for custom resolution. Hope my explanation was sufficient.

**Reply by boehm-e** [reply]
@Jirradia Ok it is very clear, I now try to install android 10 on my pi3 and will install the app you recommand :) Thank you

**Comment by boehm-e** [comment]
Hi, I encounter the same issue, I can't get Crankshaft to display on portrait mode.
I tried the basic :
```
display_rotate=1
display_hdmi_rotate=1
```

and I added 
```
echo 1 | tee /sys/class/graphics/fbcon/rotate_all
```
to /boot/crankshaft/startup.sh (which in theory rotate all screens by 90deg). It works when I write it in terminal while DEBUG is enabled.
But It does not seems to do anything when added to the startup.sh script

Is there a solution to this issue?
Thank you
